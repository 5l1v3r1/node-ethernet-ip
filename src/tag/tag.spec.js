const Tag = require("./index");

describe("Tag Class", () => {
    describe("Tagname Validator", () => {
        it("Accepts and Rejects Appropriate Inputs", () => {
            const fn = test => Tag.isValidTagname(test);

            expect(fn("_sometagname")).toBeFalsy();
            expect(fn("fffffffffffffffffffffffffffffffffffffffff")).toBeFalsy();
            expect(fn("4hello")).toBeFalsy();
            expect(fn("someTagArray[12]")).toBeTruthy();
            expect(fn("someTagArray[1a]")).toBeFalsy();
            expect(fn("hello[f]")).toBeFalsy();
            expect(fn("someOtherTag[0]a")).toBeFalsy();
            expect(fn("tagname")).toBeTruthy();
            expect(fn("tag_with_underscores45")).toBeTruthy();
            expect(fn("someTagArray[0]")).toBeTruthy();
        });
    });
});
