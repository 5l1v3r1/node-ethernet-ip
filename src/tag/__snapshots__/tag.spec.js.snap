// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Class Read Message Generator Method Generates Appropriate Buffer 1`] = `
Object {
  "data": Array [
    76,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Read Message Generator Method Generates Appropriate Buffer 2`] = `
Object {
  "data": Array [
    76,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Read Message Generator Method Generates Appropriate Buffer 3`] = `
Object {
  "data": Array [
    76,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Read Message Generator Method Generates Appropriate Buffer 4`] = `
Object {
  "data": Array [
    76,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Read Message Generator Method Generates Appropriate Buffer 5`] = `
Object {
  "data": Array [
    76,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    1,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Write Message Generator Method Generates Appropriate Buffer 1`] = `
Object {
  "data": Array [
    77,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    196,
    0,
    1,
    0,
    100,
    0,
    0,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Write Message Generator Method Generates Appropriate Buffer 2`] = `
Object {
  "data": Array [
    77,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    193,
    0,
    1,
    0,
    1,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Write Message Generator Method Generates Appropriate Buffer 3`] = `
Object {
  "data": Array [
    77,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    202,
    0,
    1,
    0,
    93,
    126,
    0,
    66,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Write Message Generator Method Generates Appropriate Buffer 4`] = `
Object {
  "data": Array [
    77,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    194,
    0,
    1,
    0,
    4,
  ],
  "type": "Buffer",
}
`;

exports[`Tag Class Write Message Generator Method Generates Appropriate Buffer 5`] = `
Object {
  "data": Array [
    77,
    3,
    145,
    3,
    116,
    97,
    103,
    0,
    195,
    0,
    1,
    0,
    246,
    255,
  ],
  "type": "Buffer",
}
`;
